pyupgrade.py

```diff
@@ -23,10 +40,29 @@
 from tokenize_rt import UNIMPORTANT_WS
 
 
+Parsed = Tuple[Text, Optional[Text], Optional[Text], Optional[Text]]
+Fmt = Tuple[
+    Optional[Text],
+    Optional[Text],
+    Optional[Text],
+    Optional[Text],
+    Optional[Text],
+]
+
+ListCompOrGeneratorExp = Union[ast.ListComp, ast.GeneratorExp]
+ListOrTuple = Union[ast.List, ast.Tuple]
+NameOrAttr = Union[ast.Name, ast.Attribute]
+
+if TYPE_CHECKING:
+    if sys.version_info >= (3,):
+        AsyncFunctionDef = ast.AsyncFunctionDef
+    else:
+        AsyncFunctionDef = ast.stmt
+
 _stdlib_parse_format = string.Formatter().parse
 
 
-def parse_format(s):
+def parse_format(s):  # type: (Text) -> Tuple[Parsed, ...]
```

Comment:
lol, not worth fixing in this branch -- but I went through all the work to preserve the types here and it's not even possible that this receives bytes in python2 now that we're using `tokenize-rt`

Reply:
Hmm I don't understand what you mean 🤣

Reply:
ignore this, I'm babbling to myself -- I'll fix it some other day